### PR TITLE
Reimplement App Navigation Menu #2345

### LINKED ISSUE #2345: Reimplement App Menu
**Actual Behaviour**

Due to the merging of repo's, a single popper is not suitable for navigation around the application.

**Expected Behaviour**

Use a better approach.

**Screenshots of the issue**

1st suggestion: 

Screenshot 2019-06-22 at 9 37 29 AM

2nd:

Screenshot 2019-06-22 at 9 37 51 AM

PS: I think 1st is better for our use case

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2345, #2388

Changes: 
- Fix scroll on IconButton Click in AppBar
- The scroll was blocked when the menu was open on the top right, fixed that as well
- Reimplemented Top Menu and Left menu connecting the application and improving the navigation.

Demo Link: https://pr-2370-fossasia-susi-web-chat.surge.sh

Screenshots of the change:

Top Menu:

Screenshot 2019-06-28 at 2 35 03 PM

Screenshot 2019-06-28 at 2 35 09 PM

Left Menu for mobile view:

Screenshot 2019-06-25 at 2 50 45 PM

### CODE REVIEW

**Comment by fragm3** [comment]
@akshatnitd @harsh-jindal please test the menu from the demo link provided.

File: src/components/StaticAppBar/TopMenu.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+class TopMenu extends React.Component {
+  static propTypes = {
+    isAdmin: PropTypes.bool,
+    accessToken: PropTypes.string,
+  };
+  render() {
+    const { isAdmin, accessToken } = this.props;
+    const navLinks = LINKS.map(link => {
+      const checkRender = ({ label }) => {
```
Comment (akshatnitd) [inline src/components/StaticAppBar/TopMenu.js]: Do we really need a function `checkRender` ?

Comment (fragm3) [reply src/components/StaticAppBar/TopMenu.js]: I could do this directly, I thought there would be more conditional links at a later stage, so we can put it in `checkRender`

**Comment by fragm3** [comment]
@akshatnitd @harsh-jindal made the changes

**Comment by mariobehling** [comment]
This is not thought-through. We need to improve, but we need to think it through first. Initially we followed the Facebook approach. You are changing this now, but there are issues:
* Some menu items are bold, others are not
* Dashboard is a - Skills Dashboard. So, it should be in the Skills Dropdown.
* The top menu items should all be clickable as well.
* You changed the underline design, which was inspired by Twitter and Google. It should be back again.

We should follow a common service. We don't have designers who can create a complete new concept for us. Which service are we following? Facebook, Google, Twitter, Amazon...? We need to think about this separately. The main goal for now is to fix the UI and get the functionality back first.

**Comment by fragm3** [comment]
Made the changes:

Screenshot 2019-06-26 at 3 01 28 AM

@mariobehling Yes I agree with you, I think we should go with Material Design practices as we are using material-ui. We need consistency(design wise) and easy navigation across the app, bridging the gap between SUSI Chat and Skills. This is a small step towards it. 

@akshatnitd @harsh-jindal please review.

**Comment by fragm3** [comment]
@akshatnitd fixed

**Comment by harsh-jindal** [comment]
@fragm3 I guess the links dashboard,botbuilder,create skill,create bot shall not be displayed when not logged in. OR open up the login modal if not logged in because currently we get page not found

**Comment by fragm3** [comment]
@harsh-jindal Done, I have hidden the links according.

**Comment by fragm3** [comment]
@akshatnitd @harsh-jindal the left menu bar drawer closes on click interaction. Please review.

**Comment by fragm3** [comment]
@samagragupta fixed.

**Comment by fragm3** [comment]
@samagragupta it is fixed in my localhost, surge not being able to deploy the new change, not sure why. Anyways, Updated link: https://sour-head.surge.sh/

**Pull request merged**